identity/handler.go

```diff
@@ -147,22 +153,31 @@ type listIdentitiesParameters struct {
 //	  default: errorGeneric
 func (h *Handler) list(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 	page, itemsPerPage := x.ParsePagination(r)
-	is, err := h.r.IdentityPool().ListIdentities(r.Context(), ExpandDefault, page, itemsPerPage)
-	if err != nil {
-		h.r.Writer().WriteError(w, r, err)
-		return
+
+	params := ListIdentityParameters{Expand: ExpandDefault, Page: page, PerPage: itemsPerPage, CredentialsIdentifier: r.URL.Query().Get("credentials_identifier")}
+	if params.CredentialsIdentifier != "" {
+		params.Expand = ExpandEverything
 	}
 
-	total, err := h.r.IdentityPool().CountIdentities(r.Context())
+	is, err := h.r.IdentityPool().ListIdentities(r.Context(), params)
 	if err != nil {
 		h.r.Writer().WriteError(w, r, err)
 		return
 	}
 
+	total := int64(len(is))
+	if params.CredentialsIdentifier == "" {
+		total, err = h.r.IdentityPool().CountIdentities(r.Context())
+		if err != nil {
+			h.r.Writer().WriteError(w, r, err)
+			return
+		}
+	}
```

Comment:
Yes, I was thinking about refactoring it. But we have to remove count anyways (see our cockroach guidlines in notion) because it's slow. So I thought it's not worth the effort if we remove this functionatily soon anyways.

Reply:
Okay, sounds good to me.